PcapDotNet/src/PcapDotNet.Core.Test/WiresharkCompareTests.cs

```diff
@@ -390,24 +391,58 @@ private static void CreateRandomIpPayload(Random random, Layer ipLayer, List<ILa
             }
         }
 
-        private static void CreateRandomUdpPayload(Random random, UdpLayer udpLayer, List<ILayer> layers)
-        {
-            if (random.NextBool(20))
-            {
-                // Finish with payload.
-                PayloadLayer payloadLayer = random.NextPayloadLayer(random.Next(100));
-                layers.Add(payloadLayer);
-                return;
-            }
-
-            DnsLayer dnsLayer = random.NextDnsLayer();
-            layers.Add(dnsLayer);
-
-            ushort specialPort = (ushort)(random.NextBool() ? 53 : 5355);
-            if (dnsLayer.IsQuery)
-                udpLayer.DestinationPort = specialPort;
-            else
-                udpLayer.SourcePort = specialPort;
+        private static void CreateRandomUdpPayload(Random random, UdpLayer udpLayer, List<ILayer> layers)
+        {
+            if (random.NextBool(20))
+            {
+                // Finish with payload.
+                PayloadLayer payloadLayer = random.NextPayloadLayer(random.Next(100));
+                layers.Add(payloadLayer);
+                return;
+            }
+
+            switch (random.Next(0, 2))
+            {
+                case 0:
+                    {
+                        DnsLayer dnsLayer = random.NextDnsLayer();
+                        layers.Add(dnsLayer);
+
+                        ushort specialPort = (ushort)(random.NextBool() ? 53 : 5355);
+                        if (dnsLayer.IsQuery)
+                            udpLayer.DestinationPort = specialPort;
+                        else
+                            udpLayer.SourcePort = specialPort;
+                        break;
+                    }
+
+                case 1:
+                    {
+                        DhcpLayer dhcpLayer = random.NextDhcpLayer();
+                        layers.Add(dhcpLayer);
+
+                        if (random.NextBool())
+                        {
+                            udpLayer.SourcePort = (ushort)(random.NextBool() ? 67 : 68);
+                        }
+                        else
+                        {
+                            udpLayer.DestinationPort = 5355;
```

Comment:
This port was copied from the DnsLayer-Test above because I thought during tests the port is randomly wrong on purpose. I just checked port 5355 and it is used for Link-Local Multicast Name Resolution. So therefor my code with random wrong port doesn't make sense. I'll remove the random port code.